crates/router/src/connector/worldpay/transformers.rs

```diff
@@ -268,7 +268,9 @@ impl TryFrom<types::PaymentsResponseRouterData<WorldpayPaymentsResponse>>
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.data.connector_request_reference_id.clone(),
+                ),
```

Comment:
Hey @speedhs 
Since Worldpay does not returns `reference id` in the response we would need to populate the worldpay's transaction_id itself.

```suggestion
               let  id = types::ResponseId::try_from(item.response.links)?;
               Ok(Self {
            status: match item.response.outcome {
                Some(outcome) => enums::AttemptStatus::from(outcome),
                None => Err(errors::ConnectorError::MissingRequiredField {
                    field_name: "outcome",
                })?,
            },
            description: item.response.description,
            response: Ok(types::PaymentsResponseData::TransactionResponse {
                resource_id: id.clone(),
                redirection_data: None,
                mandate_reference: None,
                connector_metadata: None,
                network_txn_id: None,
                connector_response_reference_id: Some(id),
            }),
            ..item.data
        })
```

You can modify the method of the `try_from` with something like this

Reply:
In this case the connector_response_reference_id is expected to take a String value as indicated by the error raised. But shouldn't it store a 'ResponseId' value like the resource_id?

Reply:
No this is for reference to locate the id in the connector dashboard and hence it cannot have a strict type so it needs to be string itself and that can be populated as I have suggested in the above comment